server/trpc/router/profile.ts

```diff
@@ -16,6 +24,43 @@ export const profileRouter = router({
       });
       return profile;
     }),
+  updateProfilePhotoUrl: protectedProcedure
+    .input(updateProfilePhotoUrlSchema)
+    .mutation(async ({ input, ctx }) => {
+      const profile = await ctx.prisma.user.update({
+        where: {
+          id: ctx.session.user.id,
+        },
+        data: {
+          image: input.url,
+        },
+      });
+      return profile;
+    }),
+  getUploadUrl: protectedProcedure
+    .input(uploadPhotoUrlSchema)
+    .mutation(async ({ ctx, input }) => {
+      const { size, type } = input;
+      const extension = type.split("/")[1];
+
+      if (!["jpg", "jpeg", "gif", "png", "webp"].includes(extension)) {
```

Comment:
making out a helper utility make sense for re-using such logic

Reply:
I'll pull it out if it becomes helpful in another part. I don't see it being used anywhere else for the moment 😁